sql/parser/builtins.go

```diff
@@ -963,15 +980,20 @@ func datumToString(datum Datum) (string, error) {
 
 var replaceSubRe = regexp.MustCompile(`\\[&1-9]`)
 
-func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+func regexpReplace(ctx EvalContext, s, pattern, to, sqlFlags string) (Datum, error) {
 	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
 	if err != nil {
 		return nil, err
 	}
 
-	patternRe, err := regexp.Compile(pattern)
-	if err != nil {
-		return nil, err
+	var patternRe *regexp.Regexp
+	if patternRe = ctx.ReCache.Lookup(pattern); patternRe == nil {
```

Comment:
I pulled the regexp compilation into the cache itself. I think Ben's right that synchronizing during the regexp compilation itself, as opposed to the lookup and update, is not worth the slowdown. For now I'm also not caching errors. We can see if we feel this is worth it later.